examples/shader/post_process_pass.rs

```diff
@@ -0,0 +1,383 @@
+    // but it's not a normal system so we need to define it manually.
+    query: QueryState<&'static ViewTarget, With<ExtractedView>>,
+}
+
+impl PostProcessNode {
+    pub const IN_VIEW: &str = "view";
+    pub const NAME: &str = "post_process";
+
+    fn new(world: &mut World) -> Self {
+        Self {
+            query: QueryState::new(world),
+        }
+    }
+}
+
+impl Node for PostProcessNode {
+    // This defines the input slot of the node and tells the render graph what
+    // we will need when running the node.
+    fn input(&self) -> Vec<SlotInfo> {
+        // In this case we tell the graph that our node will use the view entity.
+        // Currently, every node in bevy uses this pattern, so it's safe to just copy it.
+        vec![SlotInfo::new(PostProcessNode::IN_VIEW, SlotType::Entity)]
+    }
+
+    // This will run every frame before the run() method
+    // The important difference is that `self` is `mut` here
+    fn update(&mut self, world: &mut World) {
+        // Since this is not a system we need to update the query manually.
+        self.query.update_archetypes(world);
+    }
+
+    // Runs the node logic
+    // This is where you issue draw calls.
+    fn run(
+        &self,
+        graph_context: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        // Get the entity of the view for the render graph where this node is running
+        let view_entity = graph_context.get_input_entity(PostProcessNode::IN_VIEW)?;
+
+        // We get the data we need from the world based on the view entity.
+        let Ok(view_target) = self.query.get_manual(world, view_entity) else {
+            return Ok(());
+        };
+
+        // Get the pipeline resource that contains the global data we need to create the render pipeline
+        let post_process_pipeline = world.resource::<PostProcessPipeline>();
+        // The pipeline cache is a cache of all previously created pipelines.
+        // It's required to avoid creating a new pipeline each frame.
+        let pipeline_cache = world.resource::<PipelineCache>();
+
+        // Get the pipeline data for the current view
+        let Some(pipeline) = pipeline_cache.get_render_pipeline(post_process_pipeline.pipeline_id) else {
+            return Ok(());
+        };
+
+        // Get the settings uniform binding
+        let settings_uniforms = world.resource::<ComponentUniforms<PostProcessSettings>>();
+        let Some(settings_binding) = settings_uniforms.uniforms().binding() else {
+            return Ok(());
+        };
+
+        // Get the TextureView used for post processing effects in bevy
+        let post_process = view_target.post_process_write();
+
+        // The bind_group gets created each frame.
+        // It's important for this to match the BindGroupLayout defined in the PostProcessPipeline
+        let bind_group = render_context
```

Comment:
Why are we making a bind group here, and not in queue?

Reply:
I replied on discord, but I'll reply here too for anyone else. It's because post_process_write() can't be used in a queue stage. I thought I had a comment about that. I'll add one.